### ISSUE TITLE
Cannot override global Notification/Message placement

### ISSUE THREAD

**Issue by lppedd** [NG-ZORRO/ng-zorro-antd#4323]
### Reproduction link
[https://stackblitz.com/edit/angular-i7ihr3](https://stackblitz.com/edit/angular-i7ihr3)

### Steps to reproduce
Just change the location from the picker. You'll notice it doesn't take effect.

### What is expected?
The notification position should be overridable via
```
NzNotificationService#config(...);
```

### What is actually happening?
The global configuration always overwrite the inputted one in 
```
NzNotificationContainerComponent#setConfig(...) 
NzMessageContainerComponent#mergeMessageConfig(...)
````

```ts
const newConfig = (this.config = {
  ...this.config,
  ...config,
  ...this.nzConfigService.getConfigForComponent(trimComponentName(this.constructor.name))
});
```

The fix should be pretty straighforward, however it might mean **permanently overriding the global value, not only for the single call**.

----

| Environment | Info |
|---|---|
| ng-zorro-antd | 8.4.0 |
| Browser | Any |

**Comment by wendzhue** [comment]
Have you read about [this](https://ng.ant.design/docs/global-config/en#priority-of-global-configurations)?

If you want to change its position dynamically, you should use `set` of `NzConfigService`.

Check this modified version: https://stackblitz.com/edit/angular-i7ihr3-4ggx5w?file=src/app/app.component.ts

**Comment by wendzhue** [comment]
@lppedd I forgot to add a deprecation warning for this API. Could you please submit a PR for that?

**Comment by lppedd** [comment]
@wendzhue yes I had read that paragraph, and that's basically why I opened this issue.  

I think the problem is the `NzNotificationService` methods (e.g. `success`, `error`, `info`, etc) accept an `NzNotificationDataOptions` interface, which doesn't have global config's `NotificationConfig` properties. Thus you cannot really directly override global config.
Also, modifying the global config via `NzConfigService` for a single call seems dangerous, ad you always need to remember to set back the "old" value.

IDK if I explained myself very well, let me know.

**Comment by wendzhue** [comment]
1. `NotificationConfig` is going to replace `NzNotificationConfigLegacy` not `NzNotificationDataOptions`.
2. Why do you have to set back the old value? NzConfigService with merge new options with old ones. I guess I didn't get your point here.

**Comment by lppedd** [comment]
@wendzhue Mmh yeah I think we misunderstood each other.

Let's say I want all notifications to be displayed on top-center.
However for a particular usecase I need one on bottom-right.
Currently it seems the only way to do that is by changing the global configuration on-the-fly, and that doesn't feel right to me.

Is this more clear?

**Comment by wendzhue** [comment]
I am going to close this issue since #4327 was merged. Please open another issue and we can have a furthur discussion about that feature. Thank you!
